core/core_kernel.cc

```diff
@@ -29,3 +29,78 @@ long long execute_kernel_busy_wait(const Kernel &kernel)
   }
   return acc;
 }
+
+// -- add by Yuankun
+void execute_kernel_memory(const Kernel &kernel)
+{
+    //currently use two src input
+    long long jump = kernel.kernel_arg.jump;
+    long long N = kernel.kernel_arg.scratch_bytes_per_task / 3;
+
+    double *A = reinterpret_cast<double *>(kernel.kernel_arg.scratch_ptr);
+    double *B = reinterpret_cast<double *>(kernel.kernel_arg.scratch_ptr + N * sizeof(double));
+    double *C = reinterpret_cast<double *>(kernel.kernel_arg.scratch_ptr + 2 * N * sizeof(double));
+
+    for (long iter = 0; iter < kernel.iterations; iter++) {
+        for (long i = 0; i < jump; i++) {
+            for (long j = 0; j < (N/jump); j++) {
+                    long t = (i + j * jump) % (N);
+                    C[t] = A[t] + B[t];		       		 
+            }
+        }   
+    }
+    // printf("execute_kernel_memory! C[N-1]=%f, N=%lld, jump=%lld\n", C[N-1], N, jump);
+}
+
+void execute_kernel_compute(const Kernel &kernel)
+{
+
+	long long max_power = kernel.kernel_arg.max_power;
+	double temp, sum;
+	double A[128];
+
+	for (long iter = 0; iter < kernel.iterations; iter++) {
+		for (long i = 0; i < 128; i++) {
+		    temp = ((double) rand() / (RAND_MAX));
+		    sum = temp;
+		    for (long j=0; j<max_power; j++){
+		        temp *=temp;
+		        sum += temp;
+		    }    
+		    A[i] = sum;
+		}    
+	}
+	// printf("execute_kernel_memory! A[127]=%f, max_power=%lld\n", A[127], max_power);
+}
+
+void execute_kernel_io(const Kernel &kernel)
+{
+  
```

Comment:
Please `assert(false)` here until we can implement this, so people don't run it by accident.

Reply:
done